pymove/tests/preprocessing/test_filters.py

```diff
@@ -0,0 +1,567 @@
+         140460.97747220137, 7.563335999941849]
+    ], columns=cols, index=[2]
+    )
+    assert_frame_equal(move_df, expected)
+    assert move_df.len() == 1
+
+
+def test_clean_duplicates():
+    move_df, cols = _prepare_df_default(list_data_1)
+
+    duplicates_all = filters.clean_duplicates(move_df)
+    expected = DataFrame([
+        [39.984092712402344, 116.31923675537101, Timestamp('2008-10-23 05:53:05'), 1],
+        [39.984199952392578, 116.31932067871094, Timestamp('2008-10-23 05:53:06'), 1],
+        [39.984222412109375, 116.31940460205078, Timestamp('2008-10-23 05:53:11'), 2]
+    ], columns=cols, index=[0, 1, 2]
+    )
+    assert_frame_equal(duplicates_all, expected)
+    assert move_df.len() == 4
+
+    duplicates_subset = filters.clean_duplicates(move_df, subset='id')
+    expected = DataFrame([
+        [39.984092712402344, 116.31923675537101, Timestamp('2008-10-23 05:53:05'), 1],
+        [39.984222412109375, 116.31940460205078, Timestamp('2008-10-23 05:53:11'), 2]
+    ], columns=cols, index=[0, 2]
+    )
+    assert_frame_equal(duplicates_subset, expected)
+    assert move_df.len() == 4
+
+    duplicates_last = filters.clean_duplicates(move_df, subset='id', keep='last')
+    expected = DataFrame([
+        [39.984199952392578, 116.31932067871094, Timestamp('2008-10-23 05:53:06'), 1],
+        [39.984222412109375, 116.31940460205078, Timestamp('2008-10-23 05:53:11'), 2]
+    ], columns=cols, index=[1, 3]
+    )
+    assert_frame_equal(duplicates_last, expected)
+    assert move_df.len() == 4
+
+    filters.clean_duplicates(move_df, subset='id', inplace=True)
+    expected = DataFrame([
+        [39.984092712402344, 116.31923675537101, Timestamp('2008-10-23 05:53:05'), 1],
+        [39.984222412109375, 116.31940460205078, Timestamp('2008-10-23 05:53:11'), 2],
+    ], columns=cols, index=[0, 2]
+    )
+    assert_frame_equal(move_df, expected)
+    assert move_df.len() == 2
+
+
+def test_clean_nan_values():
+    move_df, cols = _prepare_df_default(list_data_1)
+    move_df.loc[3, 'id'] = nan
+
+    drop_nan = filters.clean_nan_values(move_df)
+    expected = DataFrame([
+        [39.984092712402344, 116.31923675537101, Timestamp('2008-10-23 05:53:05'), 1.0],
+        [39.984199952392578, 116.31932067871094, Timestamp('2008-10-23 05:53:06'), 1.0],
+        [39.984222412109375, 116.31940460205078, Timestamp('2008-10-23 05:53:11'), 2.0],
+    ], columns=cols, index=[0, 1, 2]
+    )
+    assert_frame_equal(drop_nan, expected)
+    assert move_df.len() == 4
+
+    filters.clean_nan_values(move_df, inplace=True)
+    expected = DataFrame([
+        [39.984092712402344, 116.31923675537101, Timestamp('2008-10-23 05:53:05'), 1.0],
+        [39.984199952392578, 116.31932067871094, Timestamp('2008-10-23 05:53:06'), 1.0],
+        [39.984222412109375, 116.31940460205078, Timestamp('2008-10-23 05:53:11'), 2.0],
+    ], columns=cols, index=[0, 1, 2]
+    )
+    assert_frame_equal(drop_nan, expected)
+    assert move_df.len() == 3
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/gh/InsightLab/PyMove/pullRequest?prid=5574968)